src/dbworker.cpp

```diff
@@ -580,10 +579,6 @@ void DBWorker::clearHistory()
 
     QList<Link> linkList;
     emit historyAvailable(linkList);
-    if (oldTabCount != 0) {
-        QList<Tab> tabList;
-        emit tabsAvailable(tabList);
-    }
```

Comment:
Running tabs should be cleared when this happens. Currently only meaningful call that ends up here is in SettingsManager. The call is triggered when you clear history from the settings.

I'd leave this over here so that tabmodel's onTabsAvailable slot can clear tabs.